functionary/core/models/user.py

```diff
@@ -94,3 +95,20 @@ def environment_permissions(
             permissions.update(self.team_permissions(environment.team))
 
         return permissions
+
+    def environments(self) -> Set["Environment"]:
```

Comment:
I think it makes sense to make this more general and have it just return the queryset.  The ordering and the fact that it's a set are kind of in service of the way you're currently using this, but at the expense of broader applicability.  Also, let's make it a property method so that the way you interact with it is similar to other model relations.

So this method becomes:

```python
@property
def environments(self) - > QuerySet:
    """docstring"""
    Environment = apps.get_model("core", "Environment")

    return Environment.objects.filter(
                models.Q(team__user_roles__user=self) | models.Q(user_roles__user=self)
            ).distinct()
````

And then later on you'd use it like:

```python
envs = self.request.user.environments.order_by("team__name", "name")

for env in envs:
   ...
```